pkg/manager/config.go

```diff
@@ -11,17 +13,28 @@ import (
 // NewConfig is used to create a new configuration object with default values.
 // Values can be overridden by passing `managercfg.Opt` options.
 func NewConfig(opts ...managercfg.Opt) (managercfg.Config, error) {
+	cfg, err := NewDefaultConfig()
+	if err != nil {
+		return managercfg.Config{}, fmt.Errorf("failed to create default manager config: %w", err)
+	}
+
+	// Override default values with the provided options.
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+	return cfg, nil
+}
+
+func NewDefaultConfig() (managercfg.Config, error) {
```

Comment:
Does it need to be public? It's used only in the above `NewConfig` 🤔

Reply:
Good call, let's expose as little as possible. :)